source/core_http_client.c

```diff
@@ -2024,39 +2070,77 @@ static HTTPStatus_t receiveAndParseHttpResponse( const TransportInterface_t * pT
 
 /*-----------------------------------------------------------*/
 
+HTTPStatus_t sendHttpRequest( const TransportInterface_t * pTransport,
+                              HTTPClient_GetCurrentTimeFunc_t getTimestampMs,
+                              HTTPRequestHeaders_t * pRequestHeaders,
+                              const uint8_t * pRequestBodyBuf,
+                              size_t reqBodyBufLen,
+                              uint32_t sendFlags )
+{
+    HTTPStatus_t returnStatus = HTTPSuccess;
+
+    assert( pTransport != NULL );
+    assert( pRequestHeaders != NULL );
+    assert( ( pRequestBodyBuf != NULL ) ||
+            ( ( pRequestBodyBuf == NULL ) && ( reqBodyBufLen == 0 ) ) );
+    assert( getTimestampMs != NULL );
+
+    /* Send the headers, which are at one location in memory. */
+    returnStatus = sendHttpHeaders( pTransport,
+                                    getTimestampMs,
+                                    pRequestHeaders,
+                                    reqBodyBufLen,
+                                    sendFlags );
+
+    /* Send the body, which is at another location in memory. */
+    if( returnStatus == HTTPSuccess )
+    {
+        if( pRequestBodyBuf != NULL )
+        {
+            returnStatus = sendHttpBody( pTransport,
+                                         getTimestampMs,
+                                         pRequestBodyBuf,
+                                         reqBodyBufLen );
+        }
+        else
+        {
+            LogDebug( ( "A request body was not sent: pRequestBodyBuf is NULL." ) );
+        }
+    }
+
+    return returnStatus;
+}
+
+/*-----------------------------------------------------------*/
+
 HTTPStatus_t HTTPClient_Send( const TransportInterface_t * pTransport,
                               HTTPRequestHeaders_t * pRequestHeaders,
                               const uint8_t * pRequestBodyBuf,
                               size_t reqBodyBufLen,
                               HTTPResponse_t * pResponse,
                               uint32_t sendFlags )
 {
-    HTTPStatus_t returnStatus = HTTPSuccess;
+    HTTPStatus_t returnStatus = HTTPInvalidParameter;
```

Comment:
I don't know why we didn't do this before. Should we also have it changed in other public methods?

Reply:
Lets do that later.